keystorage/src/main/java/tech/pegasys/web3signer/keystorage/azure/AzureConnection.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2023 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.keystorage.azure;
+
+import java.io.IOException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Map;
+
+public class AzureConnection {
+  private final HttpClient httpClient;
+
+  public AzureConnection(final HttpClient httpClient) {
+    this.httpClient = httpClient;
+  }
+
+  public Map<String, String> executeHttpRequest(final HttpRequest httpRequest) {
+
+    final HttpResponse<String> response;
+    try {
+      response = httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+    } catch (final IOException | InterruptedException | RuntimeException e) {
+      throw new RuntimeException("Error communicating with Azure vault: " + e.getMessage(), e);
```

Comment:
Will this be retryable/recoverable or crash the thread?

I'm guessing this is only executed during an external sign request sent to web3signer and so the user can always resend...will they get a 500 error in the response?

Reply:
They will get whatever is return by the azure API. When I was testing this with the worng token for instance I received a few 401 and 403.

Reply:
But in this case, we are failing to communicate with Azure, so presumably won't be able to return Azure's specific response?

Reply:
Happy to remove that if you think "Azure vault" in the message is misleading?

Reply:
I have no issue with the message, I'm just trying to understand what the UX is when this happens